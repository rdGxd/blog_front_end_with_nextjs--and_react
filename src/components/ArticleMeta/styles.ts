import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ", ";
    }

    .categories span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(50%);
    }
  `}
`;
