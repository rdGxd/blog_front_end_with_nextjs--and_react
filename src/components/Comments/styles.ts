import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
