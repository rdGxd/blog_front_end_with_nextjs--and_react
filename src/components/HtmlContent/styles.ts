import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(80%);
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      text-align: center;
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium} 0;
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
