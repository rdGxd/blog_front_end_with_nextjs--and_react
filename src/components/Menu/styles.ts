import styled from "styled-components";

export const Wrapper = styled.div``;

export const Nav = styled.nav``;

export const Logo = styled.nav``;

export const OpenClose = styled.a``;
