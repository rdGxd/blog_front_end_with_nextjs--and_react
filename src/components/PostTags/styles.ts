import styled, { css } from "styled-components";

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin: 0 0rem 0 0.5rem;
    }

    span::before {
      content: " ";
    }

    span::after {
      content: ", ";
    }

    span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(80%);
    }
  `}
`;
