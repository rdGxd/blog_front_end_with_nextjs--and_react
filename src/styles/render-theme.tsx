import { BlogThemeProvider } from "../contexts/BlogThemeContext";
import { render, RenderResult } from "@testing-library/react";

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>);
};
