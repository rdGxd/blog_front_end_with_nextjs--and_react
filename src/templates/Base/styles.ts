import styled, { css } from "styled-components";

export const Wrapper = styled.div``;

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    ${theme.spacings.xhuge}
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
