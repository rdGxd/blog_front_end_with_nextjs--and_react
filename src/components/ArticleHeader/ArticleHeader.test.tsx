import { formatDate } from "../../utils/format-date";
import { screen } from "@testing-library/react";
import { ArticleHeader } from ".";
import { renderTheme } from "../../styles/render-theme";
import mockArticleHeader from "./mock";

describe("<ArticleHeader />", () => {
  it("should render heading, excerpt, cover img and meta", () => {
    const { container } = renderTheme(<ArticleHeader {...mockArticleHeader} />);

    expect(screen.getByRole("heading", { name: mockArticleHeader.title })).toBeInTheDocument();

    expect(screen.getByRole("img", { name: mockArticleHeader.title })).toBeInTheDocument();

    expect(screen.getByText(mockArticleHeader.excerpt)).toBeInTheDocument();

    expect(screen.getByText(formatDate(mockArticleHeader.createdAt))).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
