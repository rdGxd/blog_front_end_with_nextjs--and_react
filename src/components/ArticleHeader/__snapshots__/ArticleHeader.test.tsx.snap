// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ArticleHeader /> > should render heading, excerpt, cover img and meta 1`] = `
.c5 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c5 .categories span::after {
  content: ", ";
}

.c5 .categories span:last-child::after {
  content: "";
}

.c5 a {
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c5 a:hover {
  filter: brightness(50%);
}

.c2 {
  color: #000000;
  font-size: 4.0rem;
  text-transform: none;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  margin: 2.4rem 0;
}

.c4 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      2PAPAPA
    </h1>
    <p
      class="c3"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum. Pellentesque convallis hendrerit est a auctor.
      </p>
    </p>
    <img
      alt="2PAPAPA"
      class="c4"
      src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489865/1213_78bc74760f.jpg"
    />
    <div
      class="c5"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/rodrigo-silva"
        >
          Rodrigo Silva
        </a>
        <span
          class="separator"
        >
           | 
        </span>
        <time
          datetime="2023-11-20T14:23:14.605Z"
        >
          20 de nov. de 2023
        </time>
        <span
          class="separator"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/python"
            >
              Python
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`;
